File: plugins/graphql-java-codegen-gradle-plugin/graphql-codegen-gradle-plugin/src/main/java/io/github/kobylynskyi/graphql/codegen/gradle/SchemaFinderConfig.java
package io.github.kobylynskyi.graphql.codegen.gradle;

import com.kobylynskyi.graphql.codegen.supplier.SchemaFinder;

import java.util.Collections;
import java.util.Set;

public class SchemaFinderConfig {

    private String rootDir;

    private boolean recursive = SchemaFinder.DEFAULT_RECURSIVE;

    private String includePattern = SchemaFinder.DEFAULT_INCLUDE_PATTERN;

    private Set<String> excludedFiles = Collections.emptySet();

    public String getRootDir() {
        return rootDir;
    }

    public void setRootDir(String rootDir) {
        this.rootDir = rootDir;
    }

    public boolean isRecursive() {
        return recursive;
    }

    public void setRecursive(boolean recursive) {
        this.recursive = recursive;
    }

    public String getIncludePattern() {
        return includePattern;
    }

    public void setIncludePattern(String includePattern) {
        this.includePattern = includePattern;
    }

    public Set<String> getExcludedFiles() {
        return excludedFiles;
    }

    public void setExcludedFiles(Set<String> excludedFiles) {
        this.excludedFiles = excludedFiles;
    }
}

File: plugins/sbt/graphql-java-codegen-sbt-plugin/src/main/scala/io/github/dreamylost/graphql/codegen/SchemaFinderConfig.scala
package io.github.dreamylost.graphql.codegen

import com.kobylynskyi.graphql.codegen.supplier.SchemaFinder

/**
 *
 * @author liguobin@growingio.com
 * @version 1.0,2020/7/15
 */
case class SchemaFinderConfig(
    rootDir:        String,
    recursive:      Boolean     = SchemaFinder.DEFAULT_RECURSIVE,
    includePattern: String      = SchemaFinder.DEFAULT_INCLUDE_PATTERN,
    excludedFiles:  Set[String] = Set.empty)
